pkg/cmd/roachtest/cluster.go

```diff
@@ -2666,10 +2666,18 @@ func (c *clusterImpl) StopGrafana(ctx context.Context, l *logger.Logger, dumpDir
 func (c *clusterImpl) WipeForReuse(
 	ctx context.Context, l *logger.Logger, newClusterSpec spec.ClusterSpec,
 ) error {
+	if c.IsLocal() {
+		return errors.New("cluster reuse is disabled for local clusters to guarantee a clean slate for each test")
+	}
 	l.PrintfCtx(ctx, "Using existing cluster: %s (arch=%q). Wiping", c.name, c.arch)
 	if err := c.WipeE(ctx, l, false /* preserveCerts */); err != nil {
 		return err
 	}
+	// We remove the cockroach binaries here instead of in c.Wipe as some tests
+	// rely on the assumption that cockroach binaries are not wiped in c.Wipe.
+	if err := c.RunE(ctx, c.All(), fmt.Sprintf("rm -f %s", test.DefaultCockroachPath)); err != nil {
```

Comment:
Copy and pasting from slack: "I think it should be okay after looking at the cluster reuse code. My reasoning being it should already handle the case that the home directory is empty since that should be the state of the very first cluster."

Seems to work in my limited testing. I did need to add a sudo in there to let it remove dependencies but I think that should be fine since `c.Wipe` already uses sudo. If we're concerned maybe I should kick off a 0.1 select probability test run?